springwolf-add-ons/springwolf-generic-binding/src/main/java/io/github/springwolf/addons/generic_binding/annotation/processor/PropertiesUtil.java

```diff
@@ -10,10 +10,14 @@
 import java.util.LinkedList;
 import java.util.Map;
 import java.util.Properties;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 @Slf4j
 public class PropertiesUtil {
 
+    private static final Pattern ARRAY_PATTERN = Pattern.compile("\\[([^\\]]+)\\]");
```

Comment:
I am not on a computer, so I cannot test it right now. Is it indented that the count of [ and ] do not match?

Reply:
That was the intention, but I found I can simplify it a bit more

Reply:
What do you think about this pattern `^\\[([^\\]]+)\\]$` because the square brackets are escaped and we explicitly match the beginning and the end of the string?
To avoid padded spaces at the beginning of the string, we could simply trim the string before applying the regex. 
We used this to verify the pattern:
```
[asdf,s]
[sd]
asdf[sdf]
[sdf][sdf]
[sd[sdf]]
[kdkd]dkkd
[kdkd
```

Reply:
@sam0r040 I just pushed a change adapting your proposal. With this approach only fully valid arrays are converted, anything else is kept as a String, which I think is a nice outcome.